src/main/java/org/sonatype/repository/vgo/internal/hosted/VgoHostedFacetImpl.java

```diff
@@ -0,0 +1,239 @@
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.repository.vgo.internal.hosted;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.sonatype.nexus.repository.FacetSupport;
+import org.sonatype.nexus.repository.config.Configuration;
+import org.sonatype.nexus.repository.storage.Asset;
+import org.sonatype.nexus.repository.storage.StorageFacet;
+import org.sonatype.nexus.repository.storage.StorageTx;
+import org.sonatype.nexus.repository.storage.TempBlob;
+import org.sonatype.nexus.repository.transaction.TransactionalTouchBlob;
+import org.sonatype.nexus.repository.view.Content;
+import org.sonatype.nexus.repository.view.ContentTypes;
+import org.sonatype.nexus.repository.view.Payload;
+import org.sonatype.nexus.repository.view.payloads.StreamPayload;
+import org.sonatype.nexus.repository.view.payloads.StreamPayload.InputStreamSupplier;
+import org.sonatype.nexus.transaction.Transactional;
+import org.sonatype.nexus.transaction.UnitOfWork;
+import org.sonatype.repository.vgo.VgoAssetKind;
+import org.sonatype.repository.vgo.internal.metadata.VgoAttributes;
+import org.sonatype.repository.vgo.internal.metadata.VgoInfo;
+import org.sonatype.repository.vgo.internal.util.CompressedContentExtractor;
+import org.sonatype.repository.vgo.internal.util.VgoDataAccess;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.impl.io.EmptyInputStream;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sonatype.nexus.repository.view.Payload.UNKNOWN_SIZE;
+import static org.sonatype.repository.vgo.VgoAssetKind.VGO_MODULE;
+import static org.sonatype.repository.vgo.internal.util.VgoDataAccess.HASH_ALGORITHMS;
+
+/**
+ * Vgo hosted implementation
+ *
+ * @since 0.0.2
+ */
+@Named
+public class VgoHostedFacetImpl
+    extends FacetSupport
+  implements VgoHostedFacet
+{
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  private final VgoDataAccess vgoDataAccess;
+
+  @Override
+  protected void doInit(final Configuration configuration) throws Exception {
+    super.doInit(configuration);
+  }
+
+  @Inject
+  public VgoHostedFacetImpl(final VgoDataAccess vgoDataAccess) {
+    this.vgoDataAccess = checkNotNull(vgoDataAccess);
+  }
+
+  @Override
+  @Nullable
+  @Transactional
+  public Content getInfo(final String path,
+                         final VgoAttributes vgoAttributes) {
+    checkNotNull(path);
+    String newPath = path.replaceAll("\\.info", "\\.zip");
```

Comment:
It took me a while to work out what was going on here, I wonder if we can make this more self documenting by extracting to "fetchPackageAndGenerateInfo" or something

Reply:
https://github.com/sonatype-nexus-community/nexus-repository-vgo/pull/5/commits/4d562160abb07ff648ca748aa90effaff9b08366

Reply:
I still think this is confusing "why are we getting the package zip when it was info that was request?" I wonder if extracting the second half of this method to "extractInfoFromZip" would help?

Reply:
https://github.com/sonatype-nexus-community/nexus-repository-vgo/pull/5/commits/e18e1c1fe19c64df028786c67dda5a5ba66bbbb7